Source/Core/RequestScheduler.js

```diff
@@ -0,0 +1,256 @@
+/*global define*/
+define([
+        './Check',
+        './defined',
+        './defineProperties',
+        './Heap',
+        './isBlobUri',
+        './isDataUri',
+        './RequestState',
+        '../ThirdParty/when'
+    ], function(
+        Check,
+        defined,
+        defineProperties,
+        Heap,
+        isBlobUri,
+        isDataUri,
+        RequestState,
+        when) {
+    'use strict';
+
+    /**
+     * Tracks the number of active requests and prioritizes incoming requests.
+     *
+     * @exports RequestScheduler
+     *
+     * @private
+     */
+    function RequestScheduler() {
+    }
+
+    /**
+     * The maximum number of simultaneous active requests. Un-throttled requests do not observe this limit.
+     * @type {Number}
+     * @default 50
+     */
+    RequestScheduler.maximumRequests = 50;
+
+    /**
+     * The maximum size of the priority heap. This limits the number of requests that are sorted by priority. Only applies to requests that are not yet active.
+     * @type {Number}
+     * @default 20
+     */
+    RequestScheduler.priorityHeapSize = 20;
```

Comment:
My original idea was to queue up all the requests in a frame, "sort" by most important, and then issue as many as possible to make best use of the bandwidth and our ability to process them once they are downloaded.

I actually think this can create bubbles in the requests, e.g., it would be good to make some best guest high priority requests even during the frame as requests come back in.

Is that how it works now?  I guess I will see...